account_payment_other_company/tests/test_account_payment.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (C) 2019 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _
+from odoo.tests.common import SavepointCase
+from odoo.exceptions import ValidationError
+from odoo.modules.module import get_resource_path
+from odoo.tools import convert_file
+
+
+class TestAccountPayment(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestAccountPayment, cls).setUpClass()
+        module = "account_payment_other_company"
+        convert_file(
+            cls.cr, module,
+            get_resource_path(module, "tests",
+                              "test_account_payment_data.xml"),
+            None, 'init', False, 'test', cls.registry._assertion_report,
+        )
+        cls.account_obj = cls.env['account.account']
+        cls.invoice_obj = cls.env.ref(
+            'account_payment_other_company.customer_invoice_company_a')
+        cls.invoice_company_a = cls.env.ref(
+            'account_payment_other_company.customer_invoice_company_a')
+        cls.company_a = cls.env.ref(
+            'account_payment_other_company.company_a')
+        cls.company_b = cls.env.ref(
+            'account_payment_other_company.company_b')
+        cls.account_payment_obj = cls.env['account.payment']
+        cls.journal_1 = cls.env['account.journal']
+        cls.journal_2 = cls.env['account.journal']
+        cls.account_dt1 = cls.env['account.account']
+        cls.account_dt2 = cls.env['account.account']
+        cls.account_df1 = cls.env['account.account']
+        cls.account_d21 = cls.env['account.account']
+
+        cls.company_a.due_fromto_payment_journal_id = cls.env.ref(
+            'account_payment_other_company.sales_journal_company_a'
+        )
+        cls.company_b.due_fromto_payment_journal_id = cls.env.ref(
+            'account_payment_other_company.bank_journal_company_b'
+        )
+
+        cls.company_b.due_fromto_payment_journal_id.\
+            default_debit_account_id = cls.env.ref(
+                'account_payment_other_company.a_expense_company_b'
+            )
+
+        cls.company_a.due_from_account_id = cls.env.ref(
+            'account_payment_other_company.a_recv_company_a'
+        )
+        cls.company_a.due_to_account_id = cls.env.ref(
+            'account_payment_other_company.a_pay_company_a'
+        )
+        cls.company_b.due_from_account_id = cls.env.ref(
+            'account_payment_other_company.a_recv_company_b'
+        )
+        cls.company_b.due_to_account_id = cls.env.ref(
+            'account_payment_other_company.a_pay_company_b'
+        )
+
+        cls.chart = cls.env['account.chart.template'].search([], limit=1)
+        if not cls.chart:
+            raise ValidationError(
+                # translation to avoid pylint warnings
+                _("No Chart of Account Template has been defined !"))
+
+    def test_register_payment(self):
+        self.invoice_obj.action_invoice_open()
```

Comment:
@osi-scampbell You're testing with a customer invoice, not a vendor bill.

We can have 2 tests:
* register a customer payment in the same company to make sure the default behavior continue to work
* register a vendor payment by another company

If you want to add a 3rd one, I would create a vendor bill line with no product.